File: api/app/services/stage_fields.go
package services

import (
	"sort"

	"goravel/app/models"
)

// SortStageFields sorts fields by position ascending (in place).
func SortStageFields(fields []models.StageField) {
	sort.Slice(fields, func(i, j int) bool {
		return fields[i].Position < fields[j].Position
	})
}

File: ui/src/lib/stageFields.ts
import type { StageField } from '@/api/types'

/** Returns stage fields sorted by position (ascending). */
export function sortStageFields(fields: StageField[] | undefined | null): StageField[] {
  if (!fields?.length) return []
  return [...fields].sort((a, b) => a.position - b.position)
}
